pyinteraph/path_analysis.py

```diff
@@ -23,76 +23,85 @@ def build_graph(fname, pdb = None):
     and a list of names of all the residues (graph nodes). Replaces the 
-    range indicated by the colon with all resiues in that range and 
-    keeps all residues separated by commas. Removes duplicates. Takes in
-    a string e.g. 1,3,4:56 if no PDB file is supplied.
+    range indicated by the colon with all residues in that range and 
+    keeps all residues separated by commas. Removes duplicates.
     """
-
-    # Check if PDB file is supplied
-    if pdb:
-        # Find all residues separated by commas by
-        # replacing all colon residues with ''
-        input_comma = re.sub('\w+:\w+', '', user_input)
-        # Find all residues separated by colons
-        input_colon = re.findall('\w+:\w+', user_input)
-    else:
-        # No PDB file present
-        input_comma = re.sub('\d+:\d+', '', user_input)
-        input_colon = re.findall('\d+:\d+', user_input)
-    comma_list = input_comma.split(',')
-    # Remove empty residues
-    comma_list = [res for res in comma_list if res != '']
-    # Report if any residues are not in the PDB
-    try:
-        for res in comma_list:
-            identifiers.index(res)
-    except Exception:
-        raise ValueError(f"Residue not in PDB or incorrect format: {res}")
-    colon_replace = []
-    # Substitute range of residues with the actual residues
-    for inp in input_colon:
-        try:
-            # Create list of size two with start and end of range
-            colon_split = inp.split(':')
-            # Find the index of those res in the indentifiers list
-            index = [identifiers.index(res) for res in colon_split]
-            # Replace with the residues in that range
-            inp_replace = identifiers[index[0]:index[1]+1]
-            # Concatenate to list
-            colon_replace += inp_replace
-        except Exception:
-            # Report if the specified range does not exist in the PDB
-            raise ValueError(f"Residue range not in PDB or incorrect format: {inp}")
-    # Add both lists
-    input_list = comma_list + colon_replace
+    res_list = []
+    # Split by comma and then colon to make list of list
+    split_list = [elem.split(':') for elem in user_input.split(',')]
+    for sub_list in split_list:
+        # If list size is one, word is separated by comma
+        if len(sub_list) == 1:
+            try:
+                res = sub_list[0]
+                # Find index
+                identifiers.index(res)
+                # Append to list
+                res_list.append(res)
+            except:
+                raise ValueError(f"Residue not in PDB: {res}")
+        # If list size is 2, word is separated by a single colon
+        elif len(sub_list) == 2:
+            try:
+                # Get indexes for residue in identifiers
+                index = [identifiers.index(res) for res in sub_list]
+                # Create list with all residues in that range
+                res_range = identifiers[index[0]:index[1]+1]
```

Comment:
Before putting all these under a single `try`, check that the `.index()` method cannot possibly generate an exception under these conditions, otherwise this exception could be hidden by the exception generated by the list slicing below. If this is the case, split it into multiple `try`/`except` constructs or separate the catching of the two exceptions (if they are of different types) with multiple `except` clauses.

Reply:
I'm basically only using the try statement for the `.index()` method. If it returns an error that means the residue is not in the PDB, and if it works, the following two lines should also run and I just execute them. Should I remove the `res_range` and `res_list` line from the `try` block?